### ISSUE TITLE
Reducing post processing time with demangled symbol

### ISSUE THREAD

**Issue by honggyukim** [namhyung/uftrace#598]
I found that whenever we run post processing commands such as replay, graph, etc, uftrace calls `demangle()` for every symbols. It spends the largest execution time when running such commands.

Here is the report of running `uftrace graph`.
```
$ uftrace report -s self
  Total time   Self time       Calls  Function
  ==========  ==========  ==========  ====================
  322.479 ms  251.317 ms      264560  dd_curr
  210.462 ms  210.426 ms      343296  strcmp
  283.400 ms  207.702 ms      157529  match_filter_pattern
  513.214 ms  125.323 ms       26419  dd_type
  171.761 ms  102.870 ms       15919  find_auto_args
  230.387 ms   97.212 ms       79309  __dd_consume_n
  772.073 ms   94.931 ms         304  add_trigger_entry
  145.957 ms   93.197 ms       73496  namesort
   77.916 ms   77.916 ms      289713  dd_peek
  393.741 ms   77.225 ms       16603  add_filter
  227.688 ms   69.306 ms          29  qsort
  779.505 ms   68.764 ms        4036  dd_nested_name
  252.221 ms   58.576 ms       67725  __dd_consume
   49.448 ms   49.448 ms      100238  strchr
   40.746 ms   40.746 ms      178139  dd_eof
    1.122  s   40.703 ms        4155  dd_encoding
    1.461  s   40.677 ms          13  load_module_symbol
    1.179  s   38.584 ms        8582  demangle_simple
...
```

Moreover, it takes a lot of time for handling C program symbols that do not have any symbols to be demangled.

I think we should avoid it somehow. Thanks!

**Comment by honggyukim** [comment]
The most frequent performance bottleneck would be to handle `libstdc++.so.6.0.25.sym` in my system. It is generated even when I trace C programs because uftrace itself injects it for handling full demangler.
```
$ head uftrace.data/libstdc++.so.6.0.25.sym
0000000000088fc0 P __cxa_allocate_dependent_exception
0000000000088fd0 P _ZNKSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEE16find_last_not_ofEPKwmm
0000000000088fe0 P _ITM_addUserCommitAction
0000000000088ff0 P _ZNSt8numpunctIcED2Ev
0000000000089000 P __strtof_l
0000000000089010 P _ITM_memcpyRtWn
0000000000089020 P _ZNKSt7num_putIcSt19ostreambuf_iteratorIcSt11char_traitsIcEEE13_M_insert_intIyEES3_S3_RSt8ios_basecT_
0000000000089030 P _ZNSt12strstreambufD1Ev
0000000000089040 P _ZNSt13__future_base11_State_baseD2Ev
0000000000089050 P _ZN10__cxxabiv119__pointer_type_infoD1Ev
```
Here is the recorded info.
```
$ uftrace info
# system information
# ==================
# program version     : v0.9.1-3-gfbe0 ( dwarf python tui perf sched )
# recorded on         : Fri Nov 16 20:21:08 2018
# cmdline             : uftrace record -a /home/honggyu/usr/bin/uftrace graph --no-pager
# cpu info            : Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz
    ...
```

**Comment by namhyung** [comment]
Hmm.. we may defer demangling only when it's actually used - it should reduce some work if any filter/trigger was not used at least.  Also C symbols should not enter demangler as `demangler_simple()` checks the first two character of mangled symbol first.

**Comment by namhyung** [comment]
Merged: 3dcf0f63
